plutus-ir/src/Language/PlutusIR.hs

```diff
@@ -0,0 +1,138 @@
+        PrettyBy (PLC.PrettyConfigClassic configName) (VarDecl tyname name a) where
+    prettyBy config (VarDecl _ n ty) = parens' ("vardecl" </> vsep' [prettyBy config n, prettyBy config ty])
+
+instance (PLC.PrettyClassicBy configName (tyname a)) =>
+        PrettyBy (PLC.PrettyConfigClassic configName) (TyVarDecl tyname a) where
+    prettyBy config (TyVarDecl _ n ty) = parens' ("tyvardecl" </> vsep' [prettyBy config n, prettyBy config ty])
+
+instance PrettyBy (PLC.PrettyConfigClassic configName) Recursivity where
+    prettyBy _ = \case
+        NonRec -> parens' "nonrec"
+        Rec -> parens' "rec"
+
+instance (PLC.PrettyClassicBy configName (tyname a), PLC.PrettyClassicBy configName (name a)) =>
+        PrettyBy (PLC.PrettyConfigClassic configName) (Datatype tyname name a) where
+    prettyBy config (Datatype _ ty tyvars destr constrs) = parens' ("datatype" </> vsep' [
+                                                                         prettyBy config ty,
+                                                                         vsep' $ fmap (prettyBy config) tyvars,
+                                                                         prettyBy config destr,
+                                                                         vsep' $ fmap (prettyBy config) constrs])
+
+instance (PLC.PrettyClassicBy configName (tyname a), PLC.PrettyClassicBy configName (name a)) =>
+        PrettyBy (PLC.PrettyConfigClassic configName) (Binding tyname name a) where
+    prettyBy config = \case
+        TermBind _ d t -> parens' ("termbind" </> vsep' [prettyBy config d, prettyBy config t])
+        TypeBind _ d ty -> parens' ("typebind" </> vsep' [prettyBy config d, prettyBy config ty])
+        DatatypeBind _ d -> parens' ("datatypebind" </> prettyBy config d)
+
+instance (PLC.PrettyClassicBy configName (tyname a), PLC.PrettyClassicBy configName (name a)) =>
+        PrettyBy (PLC.PrettyConfigClassic configName) (Term tyname name a) where
+    prettyBy config = \case
+        Let _ r bs t -> parens' ("let" </> vsep' [prettyBy config r, vsep' $ fmap (prettyBy config) bs, prettyBy config t])
+        Var _ n -> prettyBy config n
+        TyAbs _ tn k t -> parens' ("abs" </> vsep' [prettyBy config tn, prettyBy config k, prettyBy config t])
+        LamAbs _ n ty t -> parens' ("lam" </> vsep' [prettyBy config n, prettyBy config ty, prettyBy config t])
+        Apply _ t1 t2 -> brackets' (vsep' [prettyBy config t1, prettyBy config t2])
+        Constant _ c -> parens' ("con" </> prettyBy config c)
+        TyInst _ t ty -> braces' (vsep' [prettyBy config t, prettyBy config ty])
+        Error _ ty -> parens' ("error" </> prettyBy config ty)
+        Wrap _ n ty t -> parens' ("wrap" </> vsep' [ prettyBy config n, prettyBy config ty, prettyBy config t ])
+        Unwrap _ t -> parens' ("unwrap" </> prettyBy config t)
+
+prettyDef :: Term TyName Name a -> Doc ann
+prettyDef = prettyBy $ PLC.PrettyConfigClassic PLC.defPrettyConfigName
```

Comment:
We have

```haskell
-- | Pretty-print a value in the default mode using the classic view.
prettyPlcDef :: PrettyPlc a => a -> Doc ann
```

in `Language.PlutusCore.Pretty`.

Reply:
I don't think that's directly applicable here, unless I've misunderstood?

Reply:
Ah, yes, you're right. I kinda wish we had the `PrettyPlc` constraint everywhere, so that everything is tweakable enough, but yeah, this is fine too.